data/src/main/java/org/apache/iceberg/data/TableMigrationUtil.java

```diff
@@ -151,6 +256,36 @@ public static List<DataFile> listPartition(
     }
   }
 
+  private static List<FileStatus> fileStatusListInPartition(String partitionUri, Configuration conf)
+      throws IOException {
```

Comment:
I thought the reason you're defining this helper method is because the same code exists in the existing `listPartitions` that you're retaining but deprecating and the new one you're adding, but you haven't changed the deprecated `listPartitions` to use this helper.

Reply:
I suppose it is not worthwhile to change code you plan to remove in a later version?